docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md

```diff
@@ -0,0 +1,599 @@
+
+- Create a Markdown post
+- Add an image, and add it to the Markdown posts frontmatter
+- Query this image with GraphQL
+
+#### Creating the post
+
+```shell
+mkdir -p content/blog/2019-01-04-hello-world-seo
+touch content/blog/2019-01-04-hello-world-seo/index.md
+```
+
+```md:title=content/blog/2019-01-04-hello-world-seo/index.md
+---
+date: 2019-01-04
+featured: images/featured.jpg
+---
+
+Hello World!
+```
+
+#### Adding the image
+
+Feel free to add whatever, or perhaps use this _very pertinent_ image:
+
+![Sample Image](./images/seo.jpg)
+
+the image will need to be located at `content/blog/2019-01-04-hello-world-seo/images/featured.jpg`
+
+#### Querying with GraphQL
+
+```jsx:title=src/templates/blog-post.js
+import React from 'react'
+import { graphql } from 'gatsby'
+
+import Layout from '../components/layout'
+import SEO from '../components/seo' // highlight-line
+
+// highlight-start
+function BlogPost({ data, location }) {
+  const { markdown: { excerpt, html, fields, frontmatter } } = data
+  const image = frontmatter.image ? frontmatter.image.childImageSharp.resize : null
+  // highlight-end
+  return (
+    <Layout>
+      {/* highlight-start */}
+      <SEO title="My Amazing Gatsby App" description={excerpt} image={image} pathname={fields.slug}>
+      /* highlight-end */}
+      <h1>{frontmatter.title}</h1>
+      <div dangerouslySetInnerHTML={{ __html: html }} />
+    </Layout>
+  )
+}
+
+export const blogPostQuery = graphql`
+  # highlight-start
+  query BlogPostBySlug($slug: String!) {
+    markdown: markdownRemark(fields: { $slug: { eq: $slug }}) {
+      html
+      excerpt(pruneLength: 160)
+      fields {
+        slug
+      }
+      frontmatter {
+        image: featured {
+          childImageSharp {
+            resize(width: 1200) {
+              src
+              height
+              width
+            }
+          }
+        }
+        title
+      }
+    }
+  }
+  # highlight-end
+`
+
+export default BlogPost
+```
+
+There are a few aspects worth nothing here:
+
+1. You're using `pruneLength: 160` for the excerpt; this is because [SEO meta descriptions should be between 150-170 characters][seo-description-length]
+
+- This is a slick feature of Gatsby's GraphQL capabilities, and will truncate (e.g. with a trailing `...`) appropriately. Perfect!
+
+1. The image querying is intentionally simplified, but a good base to build upon. There are specific size and aspect ratio requirements for [both Facebook][facebook-og-image] and [Twitter][twitter-image].
+
+## The Payoff
+
+Using the techniques outlined in this post, you've made your Gatsby application SEO-friendly as well as sharable on common social networks. Check out the following examples of a sample blog post.
+
+### Google
+
+![Google](./images/google.png)
+
+### Facebook
+
+![Facebook](./images/facebook.png)
+
+### Twitter
+
+![Twitter](./images/twitter.png)
+
+### Slack
+
+![Slack](./images/slack.png)
+
```

Comment:
Please provide more specific alt text for these images. As is, it'll read as something like "Image: Google" and so on, which is not super descriptive.